pkg/vz/rosetta_directory_share.go

```diff
@@ -0,0 +1,14 @@
+//go:build darwin && !arm64
+// +build darwin,!arm64
+
+package vz
+
+import (
+	"fmt"
+
+	"github.com/Code-Hex/vz/v3"
+)
+
+func createRosettaDirectoryShareConfiguration(install bool) (*vz.VirtioFileSystemDeviceConfiguration, error) {
+	return nil, fmt.Errorf("Rosetta is unsupported on non-ARM64 hosts")
```

Comment:
We should do a warning instead of error.

This is to make vz template runnable on both arm & x64

Reply:
Couldn't users just disable Rosetta on x86 in that case?

Reply:
I also think in that case, we probably can still return an error here, but we can turn it into a named error and check using `errors.Is` in the caller.

Reply:
Updated